src/server/qgsbufferserverresponse.h

```diff
@@ -42,6 +42,12 @@ class QgsBufferServerResponse: public QgsServerResponse
 
     virtual void clearHeader( const QString& key ) override;
 
+    virtual QString getHeader( const QString& key ) const override;
+
+    virtual QList<QString> headerKeys() const override;
+
+    virtual bool headersWritten() const override;
```

Comment:
May I suggest `headersSent` instead of written? I think it's more intuitive and common among other frameworks and languages.